client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -92,18 +169,76 @@ class SpeedUpSiteSettings extends Component {
 	}
 }
 
-export default connect( state => {
-	const selectedSiteId = getSelectedSiteId( state );
-	const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
-	const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
-		state,
-		selectedSiteId,
-		'photon'
-	);
+export default connect(
+	state => {
+		const selectedSiteId = getSelectedSiteId( state );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const siteAcceleratorSupported = isJetpackMinimumVersion( state, selectedSiteId, '6.6-alpha' );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			'photon'
+		);
+		const photonModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon' );
+		const assetCdnModuleActive = isJetpackModuleActive( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnActivating = isActivatingJetpackModule( state, selectedSiteId, 'photon-cdn' );
+		const isPhotonDeactivating = isDeactivatingJetpackModule( state, selectedSiteId, 'photon' );
+		const isAssetCdnDeactivating = isDeactivatingJetpackModule(
+			state,
+			selectedSiteId,
+			'photon-cdn'
+		);
+
+		let togglingSiteAccelerator;
+		// First Photon activating.
+		if ( isPhotonActivating ) {
+			if ( assetCdnModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Asset CDN activating.
+		} else if ( isAssetCdnActivating ) {
+			if ( photonModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Photon deactivating.
+		} else if ( isPhotonDeactivating ) {
+			if ( assetCdnModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+			// Then Asset CDN deactivating.
+		} else if ( isAssetCdnDeactivating ) {
+			if ( photonModuleActive ) {
+				togglingSiteAccelerator = false;
+			} else {
+				togglingSiteAccelerator = true;
+			}
+		} else {
+			togglingSiteAccelerator = false;
+		}
 
-	return {
-		photonModuleUnavailable: siteInDevMode && moduleUnavailableInDevMode,
-		selectedSiteId,
-		siteSlug: getSiteSlug( state, selectedSiteId ),
-	};
-} )( localize( SpeedUpSiteSettings ) );
+		// Status of the main site accelerator toggle.
+		const siteAcceleratorStatus = photonModuleActive || assetCdnModuleActive ? true : false;
```

Comment:
Do you think could have the shorter:

```
photonModuleActive || assetCdnModuleActive
```

instead of:

```
photonModuleActive || assetCdnModuleActive ? true : false
````

Reply:
Ah, this could happen if both are `null`. Then an explicit boolean conversion is a viable option you can see widely used throughout Calypso:

```
!! ( photonModuleActive || assetCdnModuleActive )
```

Reply:
Fixed in 3d039eb986cd5a05d3c06064113e6a218f33f3c4